### ISSUE TITLE
Can't Renew Certificates

### ISSUE THREAD

**Issue by germansosa** [tnozicka/openshift-acme#11]
I have an issue when certificate renew runs, the route created for well-known challenge doesn't work if the main route allow insecure traffic or redirects insecure traffic, in that case, the well-known route is ignored and all traffic goes to the main route. I think this is an issue with openshift, but, we can do something to workaround this?

I am using openshift 1.5.0

**Comment by tnozicka** [comment]
@germansosa could you also provide your route, please? I want to check the route combinations and we should file a bug to OpenShift if that proves true. I thought this should work, or at least should have failed initially and not with the renewal.

**Comment by germansosa** [comment]
@tnozicka here is my route:

```
apiVersion: v1
kind: Route
metadata:
  annotations:
    kubernetes.io/tls-acme.hash: usswMS7taGDLj0mV91f+1PFYqQZmORB2J9feJv8c+oQ=
    kubernetes.io/tls-acme.last-update-time: 2017-04-10T15:41:18Z
    kubernetes.io/tls-acme: "true"
  creationTimestamp: null
  labels:
    butter.sh/letsencrypt-managed: "yes"
  name: geoip-prod
spec:
  host: geoip.servicios.cmd.com.ar
  tls:
    certificate: |
      -----BEGIN CERTIFICATE-----
      xxx
      -----END CERTIFICATE-----
      -----BEGIN CERTIFICATE-----
      xxx
      -----END CERTIFICATE-----
    insecureEdgeTerminationPolicy: Allow
    key: |
      -----BEGIN RSA PRIVATE KEY-----
      xxx
      -----END RSA PRIVATE KEY-----
    termination: edge
  to:
    kind: Service
    name: geoip-nodejs
    weight: 100
  wildcardPolicy: None
status:
  ingress:
  - conditions:
    - lastTransitionTime: 2017-03-27T18:52:00Z
      status: "True"
      type: Admitted
    host: geoip.servicios.cmd.com.ar
    routerName: router
    wildcardPolicy: None
```

This worked initially because it did not have tls enabled, the first certificate was created from a non secure route.

**Comment by andrewklau** [comment]
Yes I see this too on a few test routes, I believe it's the same issue as https://github.com/tnozicka/openshift-acme/issues/5

**Comment by tnozicka** [comment]
I will take a look but that would seem like a bug in OpenShift, just need to collect data to file it

**Comment by tnozicka** [comment]
or you guys can file it as well if you already have the data :) and cc me there

**Comment by andrewklau** [comment]
Could it be because (maybe?) openshift priorities the https route first? Perhaps it'd be possible to renew over https. Mind you, I haven't had time to fully look at the code/spec.

**Comment by andrewklau** [comment]
Testing this further, it looks like the issue relates to the mixing of http and https.

As soon as the route has https, OpenShift is ignoring any http based route.

eg.
https://www.example.com/
http://www.example.com/test/ <-- ignored
https://www.example.com/test2/ <-- accepted

I think that's the correct way to handle it, because essentially the https route due to the "Insecure Traffic:" setting of (None, Allow, Redirect) is handling both http and https.

The fix is to set the acme challenge to be TLS true with Insecure Traffic to "Allow". This will then let the router route traffic to the route. When it's TLS false (ie. http only) the openshift router is ignoring the route. Hence the reason renewals fail and why it's also not possible to have the existing route with an existing cert in the first play. 

The default openshift subdomain (even if it has a https cert) seems to be handled differently.

**Comment by tnozicka** [comment]
@germansosa sorry for the inconvenience, we are working on a fix. In case you didn't already figured it out: You can force "renewal" by deleting the secret associated with the Route, restarting acme-controller (to clear cache), and deleting the TLS spec from an route. This should trigger new certificate being issued to you.

**Comment by germansosa** [comment]
Thanks! The fix in #13 works perfect!

**Comment by andrewklau** [comment]
@germansosa can you check if your certificate actually got renewed or it just stopped erroring? I noticed a few certs didn't get updated but they passed the validation.
